### ISSUE TITLE
mock out graph API calls for better GraphConnector test coverage

### ISSUE THREAD

**Issue by ashmrtn** [alcionai/corso#1946]
There's a lot of different scenarios to test when it comes to building collections for incremental backups. These can be difficult and time consuming to setup in a real mailbox so it would be best if we could mock out the data input for this.

Broad categories of tests that would be good to check:
* expired tokens/URLs (all and some expired)
* deleted container while pulling IDs
* deleted container before starting to pull IDs but after finding it in the resolver
* container missing from the ID map
* no delta token for container
* unparsable path for container in ID map
* container move
* container rename
* container deleted before resolver finds it

**Comment by ashmrtn** [comment]
closing as duplicate of #1967
